version: "3"

services: 
  django_app:
    build: .
    volumes:
      - .:/app
    depends_on:
      - postgres
    labels:
      - traefik.enable=true
      - traefik.http.routers.django.rule=Host(`api.rrqq.me`)
      - traefik.http.routers.django.entrypoints=https
      - traefik.http.routers.django.tls=true
      - traefik.http.routers.django.tls.certresolver=letsencrypt

  postgres:
    image: postgres:12.3
    volumes:
      - postgres_data:/var/lib/postgresql/data
    environment:
      - POSTGRES_DB=seguridad
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=remon123
    

  traefik:
    image: traefik:v2.0-alpine
    ports:
      - "80:80"
      - "443:443"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - ./config/traefik/traefik.toml:/etc/traefik/traefik.toml
    # labels:
    #   - traefik.enable=true
    #   - traefik.http.routers.web-router.rule=Host(`api.rrqq.me`)
    #   - traefik.http.routers.web-router.middlewares=auth
    #   - traefik.http.routers.web-router.service=api@internal
    #   - traefik.http.routers.web-router.tls.certresolver=letsencrypt
    #   - traefik.http.routers.web-router.tls=true


volumes:
  .:
  postgres_data:
  # traefik-public-certificates: